### PR TITLE
MDE/KFE-71

### PR DESCRIPTION
**Pull request by mantvydasdeltuva**
# Initial (2024-12-20)
- [MDE/KFE-71 Updated filter value input and aggregation ui](https://github.com/mantvydasdeltuva/kath/commit/41c53bf151f6818ccc5df717631831ad4b0e41d5)
- [MDE/KFE-71 Implemented filter suffix helper function](https://github.com/mantvydasdeltuva/kath/commit/850e232a71ed5933ec89bed1d6ed5acf983acb3b)
- [MDE/KFE-71 Refactored file fetch with filters, sorts using indexing](https://github.com/mantvydasdeltuva/kath/commit/e4c10d874a86fd89536e78c5828ff0f41a070937)
- [MDE/KFE-71 Refactored file save to use indexing](https://github.com/mantvydasdeltuva/kath/commit/efa91ebfd6ccdfa476da986d4f8db8693530a7b0)
- [MDE/KFE-71 Connected filters to backend](https://github.com/mantvydasdeltuva/kath/commit/85d2f979023ea050eff8fe02c6ff182303b33b2f)
# Changes (2024-12-22)
- [MDE/KFE-71 Implemented rows check before sort](https://github.com/mantvydasdeltuva/kath/pull/9/commits/de2ca4d54f05ca6fd967f18f7c72ed384cdfec36)

### CODE REVIEW

File: app/back_end/src/routes/workspace_route.py [hunk 1]
```diff
@@ -318,107 +322,157 @@ def get_workspace_file(relative_path):
...
+        #         if header:
+        #             # Read the rows within the specified range, otherwise skip to the next row.
+        #             # Loop ends when the end row is reached or the end of the file is reached.
+        #             for i, row in enumerate(reader):
+        #                 if start_row <= i < end_row:
+        #                     paginated_rows.append(row)
+        #                 total_rows += 1
+
+        #                 if i >= end_row:
+        #                     break
```
Comment (justinnas) [inline app/back_end/src/routes/workspace_route.py:371]: Commented-out code

Comment (mantvydasdeltuva) [reply app/back_end/src/routes/workspace_route.py:371]: For future

**Review by justinnas — CHANGES_REQUESTED** [review]
After running the application, I found one bug, where the filtering stops working and the application gets stuck. Reproduction steps:
1. Open a file, preferably a smaller one with a couple of lines;
2. Sort values in any order;
3. Add any filtering.

The errors seems to be caused by zip function. Snippet from console:
```
File ".../kath/app/back_end/src/routes/workspace_route.py", line 441, in get_workspace_file
    rows, indexes = zip(
ValueError: not enough values to unpack (expected 2, got 0)
```

File contents that it was tested on:
```
hello,world
a,a
b,b
```

**Comment by mantvydasdeltuva** [comment]
Issue by [justinnas](https://github.com/mantvydasdeltuva/kath/pull/9#pullrequestreview-2518676697) is resolved with commit [MDE/KFE-71 Implemented rows check before sort](https://github.com/mantvydasdeltuva/kath/pull/9/commits/de2ca4d54f05ca6fd967f18f7c72ed384cdfec36).

**Review by justinnas — APPROVED** [review]
Works perfectly! Well done

**Pull request merged**